artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java

```diff
@@ -704,9 +708,9 @@ public List<MessageReference> cancelRefs(final boolean failed,
          final List<MessageReference> refs = new ArrayList<>(deliveringRefs.size());
          MessageReference ref;
          while ((ref = deliveringRefs.poll()) != null) {
+            metrics.addAcknowledge(ref.getMessage().getEncodeSize());
```

Comment:
This one doesnt look to be synchronized on the consumer the way the others are, worth looking at as it could mean thread safety issues corrupting the metrics updates since it does stuff like -= and ++ on the values.

Reply:
Its actually synced on lock like the others

Reply:
The others dont all seem to be synchronized on lock, e.g. the ServerConsumerImpl#acknowledge and ServerConsumerImpl#individualAcknowledge dont look to be when they are calling the same method.

Reply:
I think there is a reason why these 2 methods arent synchronised, @clebertsuconic might know? maybe I need to use Atomic Integers?

Reply:
Not sure, just seems like it would be unsafe currently as there are 2 different synchronisations at work, but neither seems to cover all the usages of this. If going to atomic updates, might be a candidate for using a FieldUpdater rather than AtomicInteger

Reply:
Ive added a fiel updaotr, its only 1 field that can be accessed by multiple threads, the rest are threadsafe